docfx_yaml/extension.py

```diff
@@ -1417,58 +1417,127 @@ def find_markdown_pages(app, outdir):
                 'href': mdfile.name.lower(),
             })
 
+def resolve_cross_reference(
+    words: List[str],
+    index: int,
+    current_word: str,
+    keyword: str,
+    visited_words: List[str],
+    hard_coded_references: Dict[str, str] = None
+) -> Optional[str]:
+    """Given a word, check if it contains a reference keyword we should convert.
 
-# Finds and replaces occurrences which should be a cross reference in the given
-# content, except for the current name.
-def convert_cross_references(content: str, current_name: str, entry_names: List[str]):
+    Args:
+        words: list of words used to check and compare content before and after `current_word`
+        index: index position of `current_word` within words
+        current_word: current word being looked at
+        keyword: reference keyword to check against
+        visited_words: list of words containing words that's been processed so far
+        hard_coded_references: Optional list containing a list of hard coded reference
+
+    Returns:
+        None if current word does not contain a reference keyword, or a string
+          that contains the converted reference.
+    """
+    if keyword in current_word:
+        if hard_coded_references and keyword in hard_coded_references:
+            if "<a" not in words[index-1] and \
```

Comment:
Will you add a comment to this? I don't understand why `<a` being in the previous word makes a difference.

Same with the check on the next line.